tests/unit/FilteredGraphTraits.cpp

```diff
@@ -696,13 +696,13 @@ BOOST_AUTO_TEST_CASE(NodePairFilteredGraphTests) {
     DotNode *Exit = Input.getNodeByName("end");
 
     for (auto &It : llvm::depth_first(InvEFG(Exit)))
-      FirstResults.push_back(It->getName().str());
+      FirstResults.push_back(It->getName());
     revng_check(FirstResults.size() == ExpectedResultsNames.size());
     for (size_t I = 0; I < FirstResults.size(); ++I)
       revng_check(FirstResults[I] == ExpectedResultsNames[I]);
 
     for (auto &It : llvm::inverse_depth_first(EFG(Exit)))
-      SecondResults.push_back(It->getName().str());
+      SecondResults.push_back(It->getName());
```

Comment:
Mmmh, can you `git blame` the commit that this?
I'm afraid it was fixing the opposite problem.

Reply:
The idea here is simple: `SecondResults` is a container of `llvm::StringRef`s (but it used to be a `std::string` container at some point in the past). `It->getName()` already returns a `llvm::StringRef`, but `.str()` makes a temporary string out of it - a view to which is then inserted into the container - leading to a hanging view.